<launch>
  <arg name="input/image" default="image_raw" />
  <arg name="input/rois" default="rois" />
  <arg name="output/traffic_light_states" default="traffic_light_states" />
  <arg name="manager" default="traffic_light_recognition_nodelet_manager" />

  <arg name="use_gpu" default="false" />
  <!-- classifier_type {hsv_filter: 0, cnn: 1} -->
  <arg name="classifier_type" default="1" if="$(var use_gpu)"/>
  <arg name="classifier_type" default="0" unless="$(var use_gpu)"/>

  <node pkg="traffic_light_classifier" exec="traffic_light_classifier_node" name="traffic_light_classifier">
    <param name="use_sim_time" value="$(env AW_ROS2_USE_SIM_TIME false)"/>
    <remap from="input/image" to="$(var input/image)" />
    <remap from="input/rois" to="$(var input/rois)" />
    <remap from="output/traffic_light_states" to="$(var output/traffic_light_states)" />
    <param name="approximate_sync" value="false" />
    <param name="classifier_type" value="$(var classifier_type)" />
    <param name="model_file_path" value="$(find-pkg-share traffic_light_classifier)/data/traffic_light_classifier_mobilenetv2.onnx"/>
    <param name="label_file_path" value="$(find-pkg-share traffic_light_classifier)/data/lamp_labels.txt"/>
    <param name="precision" value="fp16"/>
    <param name="input_c" value="3"/>
    <param name="input_h" value="224"/>
    <param name="input_w" value="224"/>
  </node>

</launch>
